src/transformers/modeling_utils.py

```diff
@@ -1277,17 +1277,32 @@ def _check_and_enable_flash_attn_2(
             )
 
         if not is_flash_attn_2_available():
-            raise ImportError(
-                "Flash Attention 2 is not available. Please refer to the documentation of https://github.com/Dao-AILab/flash-attention for"
-                " installing it. Make sure to have at least the version 2.1.0"
-            )
-        else:
             flash_attention_version = version.parse(importlib.metadata.version("flash_attn"))
-            is_flash_greater_than_2 = flash_attention_version >= version.parse("2.1.0")
-            if not is_flash_greater_than_2:
-                raise ValueError(
-                    f"You need flash_attn package version to be greater or equal than 2.1. Make sure to have that version installed - detected version {flash_attention_version}"
-                )
+            if torch.version.cuda:
+                install_message = "Please refer to the documentation of https://github.com/Dao-AILab/flash-attention to install it. Make sure to have at least the version 2.1.0."
+
+                if importlib.util.find_spec("flash_attn") is None:
+                    raise ImportError(f"The packaging flash_attn seem to be not installed. {install_message}")
+
+                flash_attention_version = version.parse(importlib.metadata.version("flash_attn"))
+                if flash_attention_version < version.parse("2.1.0"):
+                    raise ImportError(
+                        f"You need flash_attn package version to be greater or equal than 2.1. Make sure to have that version installed - detected version {flash_attention_version}."
+                    )
+                else:
+                    raise ImportError(f"Flash Attention 2 is not available. {install_message}")
+            elif torch.version.hip:
+                install_message = "Please refer to the documentation of https://github.com/ROCmSoftwarePlatform/flash-attention to install it for AMD GPUs, or use the Transformers + Flash Attention 2-compatible Dockerfile available at https://github.com/huggingface/optimum-amd/tree/main/docker/transformers-pytorch-nightly-amd-gpu-flash/Dockerfile."
+
+                if importlib.util.find_spec("flash_attn") is None:
+                    raise ImportError(f"The packaging flash_attn seem to be not installed. {install_message}")
+                flash_attention_version = version.parse(importlib.metadata.version("flash_attn"))
+                if flash_attention_version < version.parse("2.0.4"):
+                    raise ImportError(
+                        f"You need flash_attn package version to be greater or equal than 2.0.4. Make sure to have that version installed - detected version {flash_attention_version}."
+                    )
+                else:
+                    raise ImportError(f"Flash Attention 2 is not available. {install_message}")
```

Comment:
Here maybe I'd preface all error messages with "FlashAttention2 has been toggled on, but it cannot be used due to the following errors: {errors}".

I'd also link to the documentation page explaining how to install everything rather than sharing installation instructions here.

Reply:
Done, thank you!